gtk/PrefsDialog.cc

```diff
@@ -909,28 +927,73 @@ NetworkPage::~NetworkPage()
     portTag_.disconnect();
 }
 
-void NetworkPage::onPortTested(bool isOpen)
+char const* NetworkPage::getPortStatusText(PortTestStatus const status) noexcept
+{
+    switch (status)
+    {
+    case PORT_TEST_UNKNOWN:
+        return _("unknown");
+    case PORT_TEST_CHECKING:
+        return _("checking…");
+    case PORT_TEST_OPEN:
+        return _("open");
+    case PORT_TEST_CLOSED:
+        return _("closed");
+    case PORT_TEST_ERROR:
+        return _("error");
+    default:
+        return nullptr;
+    }
+}
+
+void NetworkPage::updatePortStatusText()
 {
     portLabel_->set_markup(fmt::format(
-        isOpen ? _("Port is {markup_begin}open{markup_end}") : _("Port is {markup_begin}closed{markup_end}"),
-        fmt::arg("markup_begin", "<b>"),
-        fmt::arg("markup_end", "</b>")));
-    portButton_->set_sensitive(true);
-    portSpin_->set_sensitive(true);
+        _("Status: <b>{status_ipv4}</b> (IPv4), <b>{status_ipv6}</b> (IPv6)"),
+        fmt::arg("status_ipv4", getPortStatusText(portTestStatus_[Session::PORT_TEST_IPV4])),
+        fmt::arg("status_ipv6", getPortStatusText(portTestStatus_[Session::PORT_TEST_IPV6]))));
```

Comment:
Nit: just a thought (and it's fine as is right now too) but maybe

```cpp
    auto const status_ipv4 = getPortStatusText(portTestStatus_[Session::PORT_TEST_IPV4]);
    auto const status_ipv6 = getPortStatusText(portTestStatus_[Session::PORT_TEST_IPV6]);

    portLabel_->set_markup(
        status_ipv4 == status_ipv6 ? fmt::format(_("Status: <b>{status}</b>"), fmt::arg("status", status_ipv4)) :
                                     fmt::format(
                                         _("Status: <b>{status_ipv4}</b> (IPv4), <b>{status_ipv6}</b> (IPv6)"),
                                         fmt::arg("status_ipv4", status_ipv4),
                                         fmt::arg("status_ipv6", status_ipv6)));
```

Reply:
Do `gettext()` always return the same address if `msgid` is the same?

From the [manpages](https://man7.org/linux/man-pages/man3/gettext.3.html), the closest thing I can find to an answer is this:
> The resulting string is statically allocated and must not be modified or freed.

Reply:
If I wanted to use pointers there, it would read `auto const* const status_ipv4` ;) Returning e.g. `std::string_view` should do it. Or you could also compare enum values instead.

Reply:
> Or you could also compare enum values instead.

Eh, that is true. I'm not thinking straight. 🐱

Reply:
60799de